pkg/network/tracer/tracer_test.go

```diff
@@ -1881,3 +1881,119 @@ func skipIfWindows(t *testing.T) {
 		t.Skip("test unavailable on windows")
 	}
 }
+
+const (
+	numberOfRequests   = 100
+	numberOfIterations = 200
+)
+
+// TestOpenSSLVersions setups a HTTPs python server, and makes sure we are able to capture all traffic.
+func TestOpenSSLVersions(t *testing.T) {
+	if !httpSupported(t) {
+		t.Skip("HTTPS feature not available on pre 4.14.0 kernels")
+	}
+
+	if !httpsSupported(t) {
+		t.Skip("HTTPS feature not available supported for this setup")
+	}
+
+	cfg := testConfig()
+	cfg.EnableHTTPSMonitoring = true
+	cfg.EnableHTTPMonitoring = true
+	cfg.BPFDebug = true
+	tr, err := NewTracer(cfg)
+	require.NoError(t, err)
+	initTracerState(t, tr)
+	defer tr.Stop()
+
+	addr := "127.0.0.1:8001"
+	closer, err := testutil.HTTPPythonServer(t, addr, testutil.Options{
+		EnableTLS: true,
+	})
+	require.NoError(t, err)
+	defer closer()
+
+	// Giving the tracer time to install the hooks
+	time.Sleep(time.Second)
+	client, requestFn := simpleGetRequestsGenerator(t, addr)
+	var requests []*nethttp.Request
+	for i := 0; i < numberOfRequests; i++ {
+		requests = append(requests, requestFn())
+	}
+
+	client.CloseIdleConnections()
+	assertAllRequestsExists(t, tr, requests)
```

Comment:
please reuse the same mechanism that we do for https, it's better for visibility when rereading/debugging the test
`require.Eventuallyf() 
``tr.GetActiveConnection() ` it's the same path that we use in sysprobe on the server side (`cmd/system-probe/modules/network_tracer.go`)
https://github.com/DataDog/datadog-agent/blob/main/pkg/network/tracer/tracer_test.go#L1809-L1813

Reply:
done